sql/src/stream.cc

```diff
@@ -1907,20 +1930,35 @@ void stream::_process_service_check(
     }
 
     // Processing.
-    _service_check_update << sc;
-    try {
-      _service_check_update.run_statement();
+    unsigned int str_hash(qHash(sc.command_line));
+    // Did the command changed since last time?
+    if (_cache_svc_cmd[std::make_pair(sc.host_id, sc.service_id)] != str_hash) {
```

Comment:
As the cache entry might not exist yet, you should check for the entry existence first and compare string hashes only if the entry exist.

Reply:
Same as previous one.